### ISSUE TITLE
Error in MuTauPath (PFTauDiscriminator not found)

### ISSUE THREAD

**Issue by silviodonato** [cms-sw/cmssw#29447]
After having merged #28417, we are getting this error in many workflows since CMSSW_11_1_X_2020-04-09-1100
```
----- Begin Fatal Exception 10-Apr-2020 07:00:18 CEST-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 297557 lumi: 100 event: 154881620 stream: 3
   [1] Running path 'MuTauPath'
   [2] Calling method for module PFTauSelector/'TauSkimPFTausSelectedForMuTau'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: reco::PFTauDiscriminator
Looking for module label: hpsPFTauDiscriminationByLooseMuonRejection3
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
  FUNCTION VALUE DOES NOT SEEM TO DEPEND ON ANY OF THE 3 VARIABLE PARAMETERS.
```

See https://cmssdt.cern.ch/SDT/html/cmssdt-ib/#/relVal/CMSSW_11_1/2020-04-09-1100?selectedArchs=slc7_aarch64_gcc820&selectedFlavors=X&selectedStatus=failed
@swozniewski

**Comment by silviodonato** [comment]
assign reconstruction

**Comment by silviodonato** [comment]
@swozniewski

**Comment by silviodonato** [comment]
Likely we need to migrate 
https://github.com/cms-sw/cmssw/blob/master/RecoTauTag/RecoTau/plugins/PFTauSelectorDefinition.h#L48 from `PFTauDiscriminator` to `TauDiscriminatorContainer`

**Comment by silviodonato** [comment]
urgent

this issue is breaking 11 workflows
4.29
136.73
136.787
136.799
136.811
136.823
136.835
136.855
136.867
136.879
136.891

https://cmssdt.cern.ch/SDT/html/cmssdt-ib/#/relVal/CMSSW_11_1/2020-04-12-2300?selectedArchs=slc7_amd64_gcc820&selectedFlavors=X&selectedStatus=failed

**Comment by swozniewski** [comment]
Quoting my comment which is better discussed in this issue:
https://github.com/cms-sw/cmssw/issues/29448#issuecomment-612859051

**Comment by swozniewski** [comment]
Or maybe not, as Slava stated. The problem of 136.787 at least is different: https://github.com/cms-sw/cmssw/blob/49acac1cee9f9d86c4f7223614d21b547127cd38/DPGAnalysis/Skims/python/MuTauSkim_cff.py#L77-L78 needs to be replaced by the new format which I'm currently testing

**Comment by silviodonato** [comment]
Fixed by #29465
